terratorch/tasks/segmentation_tasks.py

```diff
@@ -98,22 +99,47 @@ def __init__(
         self.tiled_inference_parameters = tiled_inference_parameters
         self.aux_loss = aux_loss
         self.aux_heads = aux_heads
-        self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+        self._model_module = None
+
+        if model_factory:  
+            self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+            self.model_builder = self._build
+        else:
+            self.model_builder = self._bypass_build
+
+        self._model_module = None 
+
         super().__init__()
+
+        self._model_module = model
+        self.model = model
```

Comment:
What happens if model is not passed as parameter? it is none, correct? does it have any implications?

Reply:
I just modified it to guarantee that or a model_factory or a model is provided:
   ```
     if model_factory:  
            self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
            self.model_builder = self._build
        elif model:
            self.model_builder = self._bypass_build
        else:
            raise Exception("Or a model_factory or a torch.nn.Module object must be provided.")
```
Which will allow the selection of the most proper builder method.